clientconn.go

```diff
@@ -133,12 +141,31 @@ func (cc *ClientConn) resetTransport(closeTransport bool) error {
 		if closeTransport {
 			t.Close()
 		}
-		newTransport, err := transport.NewClientTransport(cc.dopts.protocol, cc.target, cc.dopts.authOptions)
+		// Adjust timeout for the current try.
+		dopts := cc.dopts
+		if dopts.Timeout > 0 {
+			dopts.Timeout -= time.Since(start)
```

Comment:
I wouldn't change dopts.Timeout. It makes the logic too confusing. start.Add(dopts.Timeout) [when the timeout is set] should always be the cutoff.